nostotagging.php

```diff
@@ -201,150 +200,119 @@ public function uninstall()
 	 * Also handles the form submit action.
 	 *
 	 * @return string The HTML to output.
+	 * @throws PrestaShopException if the default shop cannot be found while not under a shop context.
 	 */
 	public function getContent()
 	{
-		// Always update the url to the module admin page when we access it.
-		// This can then later be used by the oauth2 controller to redirect the user back.
 		$admin_url = $this->getAdminUrl();
-		Nosto::helper('nosto_tagging/config')->saveAdminUrl($admin_url);
 
-		$output = '';
+		// If we are not under a shop context, redirect to the default shop.
+		// All Nosto settings are stored per shop.
+		if (_PS_VERSION_ >= '1.5' && Shop::getContext() !== Shop::CONTEXT_SHOP)
+		{
+			$shop = new Shop(Configuration::get('PS_SHOP_DEFAULT'));
+			if (!Validate::isLoadedObject($shop))
+				throw new PrestaShopException('Shop not found');
```

Comment:
If the current context is not a shop, you could ask the merchant to change it instead of displaying an error.

Displaying an exception could make him think that something went wrong on its shop.